speechbrain/dataio/audio_io.py

```diff
@@ -0,0 +1,232 @@
+    format : str
+        Container format (e.g., 'WAV', 'FLAC').
+    duration : float
+        Duration in seconds.
+    """
+    
+    def __init__(self, sample_rate, frames, channels, subtype, format):
+        self.sample_rate = sample_rate
+        self.frames = frames
+        self.num_frames = frames  # Alias for compatibility
+        self.channels = channels
+        self.num_channels = channels  # Alias for compatibility
+        self.subtype = subtype
+        self.format = format
+    
+    @property
+    def duration(self):
+        """Calculate duration in seconds."""
+        return self.frames / self.sample_rate if self.sample_rate > 0 else 0.0
+    
+    def __repr__(self):
+        return (
+            f"AudioInfo(sample_rate={self.sample_rate}, frames={self.frames}, "
+            f"channels={self.channels}, subtype='{self.subtype}', "
+            f"format='{self.format}', duration={self.duration:.2f}s)"
+        )
+
+
+def load(path, *, channels_first=True, dtype=torch.float32, always_2d=False, 
+         frame_offset=0, num_frames=-1):
+    """Load audio file using soundfile.
+    
+    Arguments
+    ---------
+    path : str
+        Path to the audio file.
+    channels_first : bool, optional
+        If True, returns tensor with shape (channels, frames).
+        If False, returns tensor with shape (frames, channels).
+        Default: True.
+    dtype : torch.dtype, optional
+        Data type for the output tensor. Default: torch.float32.
+    always_2d : bool, optional
+        If True, always return 2D tensor even for mono audio.
+        If False, mono audio returns 1D tensor (frames,) when channels_first=False.
+        Default: False.
+    frame_offset : int, optional
+        Number of frames to skip at the start of the file. Default: 0.
+    num_frames : int, optional
+        Number of frames to read. If -1, reads to the end of the file. Default: -1.
+    
+    Returns
+    -------
+    tensor : torch.Tensor
+        Audio waveform as a tensor.
+    sample_rate : int
+        Sample rate of the audio file.
+    
+    Example
+    -------
+    >>> # This is an example, actual file needed to run
+    >>> # audio, sr = load("example.wav")
+    """
+    try:
+        # Read audio file - soundfile returns (frames, channels) or (frames,) for mono
+        audio_np, sample_rate = sf.read(
+            path, 
+            start=frame_offset, 
+            frames=num_frames, 
+            dtype='float32', 
+            always_2d=always_2d
+        )
+        
+        # Convert to torch tensor
+        audio = torch.from_numpy(audio_np)
+        
+        # Convert to requested dtype
+        if dtype != torch.float32:
+            audio = audio.to(dtype)
+        
+        # Handle shape conversion
+        if audio.ndim == 1:
+            # Mono audio as 1D
+            if channels_first:
+                # Need to add channel dimension: (frames,) -> (1, frames)
+                audio = audio.unsqueeze(0)
```

Comment:
The `always_2d` parameter should take precedence over `channels_first`, because if the user asks for only a single channel, we shouldn't be adding channels just so that they can go first.